smallrye-reactive-messaging-aws-sns-1.0/src/main/java/io/smallrye/reactive/messaging/aws/sns/SnsConnector.java

```diff
@@ -0,0 +1,141 @@
+
+@ApplicationScoped
+@Connector(SnsConnector.CONNECTOR_NAME)
+public class SnsConnector implements IncomingConnectorFactory, OutgoingConnectorFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnsConnector.class);
+    static final String CONNECTOR_NAME = "smallrye-aws-sns";
+
+    @Inject
+    @ConfigProperty(name = "sns-app-public-url")
+    private Optional<String> appURL;
+
+    private Vertx vertxInstance;
+    private Executor cachedThreadPoolExecutor;
+
+    @Inject
+    public void init() {
+        LOGGER.info("Initializing Connector");
+        vertxInstance = Vertx.vertx();
+        cachedThreadPoolExecutor = Executors.newCachedThreadPool();
+    }
+
+    @PreDestroy
+    public void preDestroy() {
+        LOGGER.info("Destroying Connector");
+        Optional.ofNullable(vertxInstance).ifPresent(vertx -> vertx.close());
+    }
+
+    @Override
+    public SubscriberBuilder<? extends Message<?>, Void> getSubscriberBuilder(Config config) {
+
+        String topicName = getTopicName(config);
+        return ReactiveStreams.<Message<?>> builder().flatMapCompletionStage(message -> {
+            CompletionStage<Object> cs = CompletableFuture.runAsync(() -> {
+                //send to sns
+                send(message, topicName);
+            }, cachedThreadPoolExecutor).thenApply(x -> message);
+            return cs;
+        }).ignore();
+    }
+
+    private void send(Message<?> msg, String topicName) {
+
+        try (SnsClient snsClient = openSnsClient()) {
+            CreateTopicRequest topicCreationRequest = CreateTopicRequest.builder().name(topicName).build();
+            CreateTopicResponse topicCreationResponse = snsClient.createTopic(topicCreationRequest);
+            String topicArn = topicCreationResponse.topicArn();
+
+            PublishRequest pr = PublishRequest
+                    .builder()
+                    .topicArn(topicArn)
+                    .message((String) msg.getPayload())
+                    .build();
+            PublishResponse response = snsClient.publish(pr);
+            LOGGER.info("Message ID {}", response.messageId());
+        }
+    }
+
+    @Override
+    public PublisherBuilder<? extends Message<?>> getPublisherBuilder(Config config) {
+
+        String topicName = getTopicName(config);
+        Integer port = config.getValue("port", Integer.class);
+        boolean broadcast = config.getOptionalValue("broadcast", Boolean.class).orElse(false);
+        int initialDelay = config.getOptionalValue("initDelay", Integer.class).orElse(2000);
+        SnsVerticle snsVerticle = new SnsVerticle(getAppURL(), topicName, port);
+        vertxInstance.deployVerticle(snsVerticle);
+
+        Flowable<SnsMessage<String>> flowable = Flowable.fromCallable(() -> {
+
+            SnsMessage<String> snsMsg = null;
+            try {
+                LOGGER.trace("Polling message");
+                snsMsg = snsVerticle.pollMsg();
```

Comment:
we need to make sure we are not on the event loop here. 
maybe you can reuse the executor.

Reply:
I will use Schedulers.single() I can't combine it with executor.